### PR TITLE
change graph lib, add zoom

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/1901

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 1]
```diff
@@ -1,102 +1,441 @@
...
+                        const dimension = context.dataset.yAxisID === 'y1' ? 'm' : 'm/c'
+                        if (context.parsed.y !== null) {
+                            label += `${context.parsed.y} ${dimension}`;
+                        }
+                        return label;
+                    }
+                }
+            },
+            hover: {
+                mode: "nearest",
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:275]: use single quotes

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 2]
```diff
@@ -1,102 +1,441 @@
...
+                    y2: {min: minSpeed - 10, max: maxSpeed + 10},
+                },
+                zoom: {
+                    wheel: {
+                        enabled: true,
+                        rangeMax: {
+                            y: maxEle
+                        }
+                    },
+                    mode: "xy",
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:294]: single quotes

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 3]
```diff
@@ -1,102 +1,441 @@
...
+                    speed: 100,
+                },
+                pan: {
+                    enabled: true,
+                    mode: "xy",
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:299]: single quotes

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 4]
```diff
@@ -1,102 +1,441 @@
...
+                    }
+                }
+            },
+            y2: {
+                display: showY2Scale(),
+                type: 'linear',
+                position: 'right',
+                title: {
+                    display: true,
+                    text: 'speed in m/c',
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:340]: m/c or m/s ?

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 5]
```diff
@@ -1,102 +1,441 @@
...
+        }
+    };
+
+    const graphData = {
+        labels: data.map((d) => {
+            if (d[xAxis] !== 0) {
+                return d[xAxis].toFixed(2)
+            } else {
+                return 0;
+            }
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:357]: one-liner ?

(d) => d[xAxis]!==0 ? d[xAxis].toFixed(2) : 0

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 6]
```diff
@@ -1,102 +1,441 @@
...
+                yAxisID: 'y2',
+                pointRadius: 0
+            },
+        ],
+    };
+
+    function ThumbComponent(props) {
+        const {children, className, ...other} = props;
+        const extraClassName =
+            other["data-index"] === 0 ? "first-thumb" : "second-thumb";
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:400]: single quotes

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 7]
```diff
@@ -1,102 +1,441 @@
...
+        return (
+            <SliderThumb {...other} className={clsx(className, extraClassName)}>
+                {children}
+            </SliderThumb>
+        );
+    }
+
+    function valueLabelFormat(value) {
+        return `${(data[value][xAxis]).toFixed(1)}  km`;
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:409]: one space or doub-space '  km' ?

File: map/src/infoblock/components/graph/GpxGraphProvider.jsx [hunk 8]
```diff
@@ -31,12 +53,21 @@ const GpxGraphProvider = ({width}) => {
...
+                [ELEVATION]: data.ele ? data.ele : '',
+                [ELEVATION_SRTM]: data.srtm ? data.srtm : '',
+                [SPEED]: data.speed ? data.speed : ''
+            })
+        }
+    }, [data]);
+
     const graphData = useMemo(() => {
         if (!_.isEmpty(data?.data)) {
             let elevation = data.ele ? "ele" : null;
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraphProvider.jsx:73]: single quotes

File: map/src/context/TracksManager.js [hunk 9]
```diff
@@ -136,12 +136,22 @@ function prepareLocalTrack(track) {
...
+        analysis: prepareAnalysis(prepareTrack.analysis),
         selected: false,
-        originalName: prepareTrack.originalName
+        originalName: prepareTrack.originalName,
+        hasGeo: prepareTrack.hasGeo,
     };
 }
 
+function prepareAnalysis(analysis) {
+    let newAnalysis = Object.assign({}, analysis);
```
Comment (RZR-UA) [inline map/src/context/TracksManager.js:147]: use const

File: map/src/infoblock/components/InformationBlock.jsx [hunk 10]
```diff
@@ -105,9 +103,7 @@ export default function InformationBlock({hideContextMenu, drawerWidth}) {
                         <TabContext value={value}>
                             <AppBar position="static" color="default">
                                 <div style={{display: 'inherit'}}>
-                                    <Button key='close' onClick={() => {
-                                        closeContextMenu()
-                                    }}>
+                                    <Button key='close' onClick={() => setShowContextMenu(false)}>
```
Comment (RZR-UA) [inline map/src/infoblock/components/InformationBlock.jsx:106]: use double-quotes for element attributes

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 11]
```diff
@@ -1,102 +1,441 @@
...
+                    title: (context) => {
+                        return `${xAxis}: ${context[0].label} km`
+                    },
+                    label: (context) => {
+                        let label = context.dataset.label || '';
+
+                        if (label) {
+                            label += ': ';
+                        }
+                        const dimension = context.dataset.yAxisID === 'y1' ? 'm' : 'm/c'
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:266]: m/c or m/s ?

File: map/src/infoblock/components/graph/GpxGraph.jsx [hunk 12]
```diff
@@ -1,102 +1,441 @@
...
+            </SliderThumb>
+        );
+    }
+
+    function valueLabelFormat(value) {
+        return `${(data[value][xAxis]).toFixed(1)}  km`;
+    }
+
     return (<>
             <Typography component={'span'} type="title" color="inherit" sx={{p: 0}}>
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraph.jsx:413]: is really need to call {} block to return just 'span' ? component="span" ?

File: map/src/infoblock/components/graph/GpxGraphProvider.jsx [hunk 13]
```diff
@@ -91,23 +124,67 @@ const GpxGraphProvider = ({width}) => {
...
     }, [data]);
 
+    function checkShowData(value) {
+        return value === '' ? false : value;
+    }
 
     return (<>
-            {graphData &&
+            {ctx.currentObjectType === ctx.OBJECT_TYPE_LOCAL_CLIENT_TRACK &&
+                <div style={{marginLeft: '15px', marginTop: '-10px'}}>
```
Comment (RZR-UA) [inline map/src/infoblock/components/graph/GpxGraphProvider.jsx:144]: try

style => sx
marginLeft => ml
marginTop => mt

should work good with div

File: map/src/infoblock/components/tabs/GeneralInfoTab.jsx [hunk 14]
```diff
@@ -30,6 +30,7 @@ export default function GeneralInfoTab({width, setShowContextMenu}) {
     return (<>
             <Box minWidth={width} maxWidth={width}>
                 <GeneralInfo width={width} setOpenDescDialog={setOpenDescDialog}/>
+                <Divider sx={{mt: "3px", mb: "12px"}}/>
```
Comment (RZR-UA) [inline map/src/infoblock/components/tabs/GeneralInfoTab.jsx:33]: single quotes in js block

**Pull request merged**